### ISSUE TITLE
Restrict search for album and/or artists

### ISSUE THREAD

**Issue by vprigent** [KeitIG/museeks#230]
When you right click on a track, you can search for the name of the artist and for the album of the track.

Even if it looks good on first sight, I've experienced some case where the filter is too broad. 
Searching for the artist "Her" gives me 197 track over 3200 for example.

We should perhaps look for a way to specify on which field I want to search ?
Or open a dropdown that allows us to specify what kind of thing  we are looking for ? ( Kind of I'm looking for an artist, an album, a track or i'm feeling lucky)

**Comment by KeitIG** [comment]
Hello,

Tanks for opening this, this is a known issue. Mostly due to the fact that we don't séparateur albums/artists on database.

The idea of filtering by album/artist would improve it, but not fix it. "The Do" would show "The Doors" too and so on.

The implemation/improvement of this is open to discussion, I'll post my ideas here when I have time.

**Comment by vprigent** [comment]
You are right.
And as a drowback, restrictive search for "The Do" would not give us "The Do ft IDK"

I should take a look at how are handled tracks ..

**Comment by KeitIG** [comment]
The main problem is we use NoSQL and a Track collection: there is no Album collection or Artists collection.

That's basically the main problem.

**Comment by vprigent** [comment]
However, we save the artist, the album, it should be possible to search based on these keys no ?

**Comment by KeitIG** [comment]
Yes of course, we could (and I will implement it), but two albums can have the same name. So we would need to filter by artist AND album, and then, there's two albums with the same name "greatest hits" for example, and then we're screwed, and then we cry.

The best idea would be to index albums in database as well as artists, so we could loop up for "the album with this _id".

But it'd get a bit complicated with NoSQL and the current state of the app.

**Comment by KeitIG** [comment]
Imagine we had the ORM + models system of Laravel, it'd be extremly powerful (and this issue would become a trivial problem).

Maybe we need a database with SQL and joins, but this'd a gigantic work.

**Comment by vprigent** [comment]
Hum ... I see ... Search queries become much more complex than I thought ^^

**Comment by dkniffin** [comment]
Putting aside how we implement it for a sec, I think we should support a search syntax similar to google, github, slack, and many others. So, in our case, you'd be able to do something like artist:"AC/DC" album:"Back in Black". Then in the contextmenu, we'd just have to put the right syntax in the search bar.

**Comment by vprigent** [comment]
I'm not so sûr about that. It's great if you are a developer or at least someone who has advanced knowledge on computer but most of the users does not want to have to specify what they are looking for, as in google.

It can be a start; but doing some kind of search engine using weight depending on columns would be accurate enough and easier for the final user

**Comment by dkniffin** [comment]
Hmm, that's a good point about most users being dumb :P Maybe we add a little UI, like google does, that allows you to put in Artist/Album/Song or whatever, and it updates the search bar to reflect that.

**Comment by b3u** [comment]
@martpie I think that advanced search syntax is a great idea, even if there's no UI. Also, this issue is almost four years old, bumping for visibility.
